magento/magento2
# Fix webapi product custom attribute preprocessor execution

### Description
This fixes the issue related to Webapi product custom attribute preprocessors described in the following issue:

https://github.com/magento/magento2/issues/38633

Basically this fixes the validation mechanism that checks which preprocessor should run.

This also updates some function names and interface parameters, which is related to the issue fix.

### Fixed Issues
1. Fixes magento/magento2#38633

### Manual testing scenarios
1. Inject an processor object, using "customAttributePreprocessors" array in ServiceInputProcessor class.
2. The processor injected to "customAttributePreprocessors" array should implement PreprocessorInterface interface. (Example bellow)
3. Clear cache / generated code.
4. Send a POST request to product API endpoint, in order to create/update some product data.
5. Include `manufacturer` value inside `custom_attributes` data
6. Processor `process` function should execute the desired code. (IE: `die('manufacturer: it works');`

```php
<?php

declare(strict_types=1);

namespace Vendor\Module\Webapi\CustomAttribute;

use Magento\Framework\Webapi\CustomAttribute\PreprocessorInterface;

class TestProcessor implements PreprocessorInterface
{
    public function shouldBeProcessed(string $key, $attribute): bool
    {
        return true;
    }

    public function process(string $key, &$attribute)
    {
        // Do something with the attribute
        die('manufacturer: Do something with the attribute here');
    }

    public function getAffectedAttributes(): array
    {
        return ['manufacturer'];
    }
}
```

### Contribution checklist
